### DISCUSSION TITLE
G7 Mismatch wait/Scanning Errors

### DISCUSSION THREAD

**Discussion by rpm5099** [NightscoutFoundation/xDrip#3779] [Bug reports]
I'm stuck on Mismatch - wait/scanning errors. Last sensor I ended up reinstalling about 5 times but reloading the settings would cause it to not connect, so I recreated all of the settings and it finally worked. This time, even after doing that I'm still stuck with the same status.  I'm on nightly build from Nov 17th 2024, have configured all of the recommended settings and scanned the qr code. Samsung galaxy s23 ultra is the primary and the spare phone used to check is a Oneplus Nord N30.  I took off work today to try to get this working because my daughter and I both use G7's with AAPS.  Somehow my daughters phone has not so far had this issue, but it's the only one.

Here are the steps taken, the end results is ALWAYS the same:

- Completely removed xdrip entirely and reinstalled latest nightly version
- Have tried doing no configuration except the sensor setup to connect to the G7 (including scanning qr code), as well as restoring all of my old settings.
- Testing with both phones was done with the other phone completely powered off and xdrip uninstalled
- Old sensors are in electrostatic bags and out of range
- Left my phone for ~12 hours, it failed to connect with the same errors the entire night.

I had this problem 10 days ago and somehow was able to get it paired after reinstalling xdrip and manually recreating all of my settings in the gui.  This time even that is not working.  The only thing I have ***not*** tried is discarding this new G7 sensor and trying another one, because that seems very unlikely...but then again I don't see a ton of posts with the mismatch error.

Normally my configuration consists of xdrip connecting directly to my G7, and xdrip variant 1 connecting to my daughters as a follower.  For troubleshooting I completely removed all variants, and the spare phone has almost nothing installed on it as it is not in use.  

What exactly does the "Mismatch" error mean, what is mismatching?  The pairing code or timestamp?  Is this xdrip rejecting the response from the sensor or vice versa?  

Thanks again.

![Screenshot_20241118_135123_xDrip+](https://github.com/user-attachments/assets/85438e57-0293-409b-8206-0da9d02c0338)
![Screenshot_20241118_135117_xDrip+](https://github.com/user-attachments/assets/459d95fa-6777-42fc-a5f2-be2d87ab5779)

**Comment by Navid200** [comment]
An electrostatic bag does not block Bluetooth or other electromagnetic waves.
Either move them really far or use a metallic box.

What is your phone brand and model?
What is your daughter's phone brand and model?

**Comment by rpm5099** [comment]
Update 

The old sensors were two floors away, way too far to have any signal, but now they're in a safe.  When I said electrostatic I meant faraday bag.

I did not see this sensor on the BLE Scanner and the Dexcom app doesn't find it either.  Based on the messages that show in xdrip it led me to believe that there was two way communication via bluetooth of some kind with the sensor.  When I saw connecting change to authorizing followed by mismatch the logical assumption was that the sensor was communicating and something is going wrong.

I installed another sensor and activated it with the Dexcom app successfully.  I see it showing up in the BLE scanner periodically.  Uninstalled Dexcom, installed xdrip, did all the configuration - same exact error.  Mismatch wait and scanning errors.  

I restarted the phone opened xdrip waited for it to connect.  At the same time i saw the new sensor appear in BLE scanner on the other phone the status changed to Connect request 15:17/bluetooth link- connecting/brain state - power connect and it's been stuck there now for about 10 minutes even though I see the sensor becoming active on the BLE Scanner multiple times.  I tried unpairing the sensor from bluetooth so that xdrip could handle pairing it again now that the dexcom app is removed, that made no difference.

Sorry, I'm a bit frustrated, this is really taking up a lot of time at this point.


Samsung Galaxy S23 Ultra - mine
Oneplus 6T - daughter  (no dexcom app available)
Oneplus N30 - spare (no dexcom app available)

Adding in - xdrip has not successfully paired the new (or old) sensor so it shows up in bluetooth devices.  After restarting again now it's stuck in Authorizing 2/disconnected/Initializaign errors 4...

**Reply by Navid200** [reply]
Dexcom app is a turn-key solution.  If you use it, you can be sure that it works or if it doesn't, you can contact Dexcom and they have to provide an answer.

We have no way if testing all brands and models.  The fact that Samsung effectively run their own version of Android does not help.

I have been using xDrip with G7 using Pixel 6a for 17 months with no issues.
I can try to help.  But, I cannot make any promises and if you expect a guaranteed solution, you will be unpleasantly surprised.  xDrip for obvious reasons cannot be a turn-key solution as it supports so many different data sources.

Post a screenshot of the following page:
Settings > Dex Debug Settings

**Reply by rpm5099** [reply]
> Dexcom app is a turn-key solution.  If you use it, you can be sure that it works or if it doesn't, you can contact Dexcom and they have to provide an answer.
> 
> We have no way if testing all brands and models.  The fact that Samsung effectively run their own version of Android does not help.
> 
> I have been using xDrip with G7 using Pixel 6a for 17 months with no issues.
> I can try to help.  But, I cannot make any promises and if you expect a guaranteed solution, you will be unpleasantly surprised.  xDrip for obvious reasons cannot be a turn-key solution as it supports so many different data sources.
> 
> Post a screenshot of the following page:
> Settings > Dex Debug Settings 

![Screenshot_20241118_163502_xDrip+.jpg](https://github.com/user-attachments/assets/1f886ad2-5c2f-4d4d-ae9d-9ce0d7725afc)

**Reply by rpm5099** [reply]
@Navid200 - I fully understand and appreciate all of that, and in particular your efforts to help provide support for issues on an open source project like this.  The Dexcom alternative is...unthinkable.  I have also successfully used G7 sensors with this phone in the past without issues.  TBH I don't really care about reinstalling xdrip 10 times, what I fear is manually re-entering them all via the gui because restoring old settings somehow brings back the connection issue with it. And here I am talking about settings were just used successfully for 10 days with the previous G7 sensor.  It seems like only a *clean* install seems to work, which doesn't strike me as an issue with the phone, but I will be switching to pixel for my next phone.  I strongly considered it last time around.

I don't know if the sensor from last night was bad, or if it sat without successfully getting activated by *something* it shutdown after some number of hours - is that a thing?  In the future I will be sure to activate it with the dexcom app if necessary to avoid trashing a perfectly good sensor.

Thanks again for your help, I'm willing to try to figure this out for my self but also anyone else who is experiencing the issue.  I've read all of the discussions and issues that have g7 and mismatch in them, and nothing jumped out at me.

**Reply by Navid200** [reply]
I restore my settings with no issue.  But, I am certain the saved settings are correct and best. 
The best settings with respect to intermittent connectivity vary from phone to phone: https://navid200.github.io/xDrip/docs/Intermittent.html

**Comment by rpm5099** [comment]
I was able to get the sensor to activate and connect to xdrip on the spare Oneplus Nord N30 phone, and it did send back BG readings.  On the galaxy s23 ultra the situation remains the same no matter what I do.  I cannot seem to figure out what the problem is.


Edit:  After reinstalling xdrip once again on the S23 and configuring the G7 manually without restoring my settings it finally connected after about half an hour of showing errors.  I'm too worried about screwing it up by reloading my settings so I put in the bare minimum and left it.  I'm very concerned about having to deal with this again in 10 days...

**Reply by Navid200** [reply]
Just to let you know in case it may be useful, you can restore the database from a previous installation without restoring the settings.  

See how it goes now.  If it disconnects, you may want to try enabling special pairing workarounds.

Regardless, you need to disable battery optimization of xDrip of any sort.   I believe there is a setting on Samsungs under device care.

In 10 days, this is what I do:
After inserting the new sensor, I use another phone and establish connectivity to ensure it is working fine.  I then disable collection on this phone.
When time comes to switch from my old sensor to the new one, I peel off the old sensor and put it away such that it cannot be seen by my phone.
I then delete all devices starting with DXCM.  Then, I change the transmitter ID to the new pairing code and wait for pairing.
After pairing, I restart the phone.

**Reply by rpm5099** [reply]
I'm pretty much like you in terms of the database - it's all saved in nightscout so I don't care too much about losing that.  It's the settings that are painful to re-enter manually.  I will do exactly as you've described here in 10 days and reply on this discussion.  If you think of anything else please let me know.  Thanks again.

**Reply by Navid200** [reply]
G7 goes into rapid reconnect mode after 15 minutes of not having connectivity.  
That's the only time you can get a pairing request.
What I mean is that if you have a G7 that is already paired with an app and has had a recent handshake, another xDrip will never see it and cannot be confused by it.

On the other hand, if you want xDrip to see it and pair with it, not only you need to disable the other app, but you also need to wait 15 minutes.  In the meantime, xDrip will be looking and getting no pair requests.  

My hope is that you will not need to keep reinstalling.  I haven't had to do that for a very long time.

**Reply by rpm5099** [reply]
Ok, this make sense as to the delay after activating it on the spare phone.  


One setting I had never actually set was the Bluetooth battery settings - on Samsung you need to go into Settings->Apps then click on the search button at the top and enable showing system apps.  I set the battery for "Bluetooth", "Bluetooth Agent" and "Bluetooth MIDI Service" (probably unrelated) to "Unrestricted" from "Optimized".  It seems like the app itself having unrestricted background and bluetooth permissions could use bluetooth all it wants but maybe not.

**Reply by Navid200** [reply]
With Pixel, it is essential to disable battery optimization for Bluetooth as well.
Having learnt that lesson the hard way, and having seen it advised by other Pixel users as well, I have chosen to make that advice, in the guide, regardless of brand.  
Without it, Pixel also suffers from intermittent connectivity.

**Comment by rpm5099** [comment]
Ok, have an update.  The sensor fell off this morning (day 9), so I put it in an RF shielded container inside a safe.  I inserted the new sensor and started the activation on the main phone instead of the spare because the last sensor was already gone.  I unpaired the old sensor in the bluetooth settings and entered the code for the new sensor.  Initially it got some sort of connection right away and approved the pairing request but it showed sensor status as stopped, bluetooth connected, brain deep sleeping errors, so I waited about 10 minutes with no change.  Based on your advice I then restarted the phone and opened it right back up to the status page.  At first it looked like the sensor had never connected - bluetooth not showing, phone/brain scanning.  At some point it changed to service state connect request, bluetooth connecting and brain power connect.  After a long minute or two I got sensor status warming up.  

So it worked, but not without scaring me that I was going to spend the whole day on it again :cold_sweat:.  I'm including shots of the statuses in case it provides any insight to issues with the G7 on this device.  Although it sounds like dev work for bluetooth with the G7 on this version of Android has stopped, obviously the priority is the new version of Android.  Thanks again for your help @Navid200, happy Thanksgiving if you are in the US!

![xdrip_g7_1](https://github.com/user-attachments/assets/774d0738-8d7f-4921-9956-35a9f2cd620c)
![xdrip_g7_2](https://github.com/user-attachments/assets/ac06043b-21c1-4903-b7aa-ffc542b06768)
![xdrip_g7_3](https://github.com/user-attachments/assets/f5bb6709-9417-409d-89c7-33b0a338841b)
![xdrip_g7_4](https://github.com/user-attachments/assets/b170d67b-4ef4-4391-b501-5fd92b547a2c)

**Reply by Navid200** [reply]
Thanks a lot for the report.
Development has not stopped.
I am monitoring all of this and hopefully will have a submission soon to make an improvement.

**Comment by cstromblad** [comment]
I have nothing to add  beyond that I've just experienced this error myself. I decided to discard the sensor and try another one, which worked. I seem to have gotten a really bad batch of sensors because only about a week ago one suddenly stopped working after only a few days. And just had to replace another one where the glue failed after 4 days.

But two sensors, from the same batch, has failed due to some technical errors.
